### ISSUE TITLE
[bug] build_folder_vars don't work on Windows with single config generators

### ISSUE THREAD

**Issue by WernerWessely** [conan-io/conan#12918]
### Environment details

* Operating System+version: Windows 10
* Compiler+version: Visual Studio 16
* Conan version: 1.57.0
* Python version: 3.11.0

### Steps to reproduce

1. Create:
`conan new -m cmake_lib lib/0.1.0 `
2. Install:
`conan install -if build -c:h tools.cmake.cmaketoolchain:generator=Ninja -c:h tools.cmake.cmake_layout:build_folder_vars="['settings.os']" .`
3. Build:
`conan build -if build .`
4. Take a look at build/windows. Since Ninja is single config, I would expect the project to be configured and built into build/windows/Release (or Debug) but it is configured and built directly into build/windows. The same happens when using the 'MinGW Makefiles' generator.
If I run the same steps under Linux with 'Unix Makefiles' or 'Ninja', the behavior is as expected.

### Logs

```
C:/lib ➜ conan new -m cmake_lib lib/0.1.0
File saved: CMakeLists.txt
File saved: conanfile.py
File saved: include/lib.h
File saved: src/lib.cpp
File saved: test_package/CMakeLists.txt
File saved: test_package/conanfile.py
File saved: test_package/src/example.cpp
```
```
C:/lib ➜ conan install -if build -c:h tools.cmake.cmaketoolchain:generator=Ninja -c:h tools.cmake.cmake_layout:build_folder_vars="['settings.os']" .
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
[build_requires]
[env]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
tools.cmake.cmake_layout:build_folder_vars=['settings.os']

conanfile.py (lib/0.1.0): Installing package
Requirements
Packages

Installing (downloading, building) binaries...
conanfile.py (lib/0.1.0): Generator txt created conanbuildinfo.txt
conanfile.py (lib/0.1.0): Calling generate()
conanfile.py (lib/0.1.0): WARN: Using the new toolchains and generators without specifying a build profile (e.g: -pr:b=default) is discouraged and might cause failures and unexpected behavior
conanfile.py (lib/0.1.0): Preset 'windows-release' added to CMakePresets.json. Invoke it manually using 'cmake --preset windows-release'
conanfile.py (lib/0.1.0): If your CMake version is not compatible with CMakePresets (<3.19) call cmake like: 'cmake <path> -G Ninja -DCMAKE_TOOLCHAIN_FILE=C:\lib\build\windows\generators\conan_toolchain.cmake -DCMAKE_POLICY_DEFAULT_CMP0091=NEW -DCMAKE_BUILD_TYPE=Release'
conanfile.py (lib/0.1.0): Aggregating env generators
conanfile.py (lib/0.1.0): Generated conaninfo.txt
conanfile.py (lib/0.1.0): Generated graphinfo
```
```
C:/lib ➜ conan build -if build .
Using lockfile: 'C:\lib\build/conan.lock'
Using cached profile from lockfile
conanfile.py (lib/0.1.0): Calling build()
conanfile.py (lib/0.1.0): CMake command: cmake -G "Ninja" -DCMAKE_TOOLCHAIN_FILE="C:/lib/build/windows/generators/conan_toolchain.cmake" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "C:\lib\."
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.11
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
-- Using Conan toolchain: C:/lib/build/windows/generators/conan_toolchain.cmake
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The CXX compiler identification is MSVC 19.29.30141.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_POLICY_DEFAULT_CMP0091


-- Build files have been written to: C:/lib/build/windows
conanfile.py (lib/0.1.0): CMake command: cmake --build "C:\lib\build\windows" -- -j12
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.11.11
** Copyright (c) 2021 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
[2/2] Linking CXX static library lib.lib
```

**Comment by memsharded** [comment]
Hi @WernerWessely 

Thanks for your report, it is clear. Indeed doesn't look good, I will try to have a look and fix it asap.

**Comment by memsharded** [comment]
Fixed in #12925, will be in next 1.58 (or we might consider doing a minor 1.57.1, we will talk about it).
Thanks again for reporting.

**Comment by WernerWessely** [comment]
> 

NP, and thanks for your great work!
